core/iwasm/common/wasm_runtime_common.c

```diff
@@ -1111,21 +1111,32 @@ wasm_runtime_load(uint8 *buf, uint32 size, char *error_buf,
                   uint32 error_buf_size)
 {
     WASMModuleCommon *module_common = NULL;
+    WASMModule *module = NULL;
 
     if (get_package_type(buf, size) == Wasm_Module_Bytecode) {
 #if WASM_ENABLE_INTERP != 0
         module_common =
             (WASMModuleCommon *)wasm_load(buf, size, error_buf, error_buf_size);
-        return register_module_with_null_name(module_common, error_buf,
-                                              error_buf_size);
+        module_common = register_module_with_null_name(module_common, error_buf,
+                                                       error_buf_size);
+
+        if (module_common == NULL) {
+            return NULL;
+        }
+        goto success;
 #endif
     }
     else if (get_package_type(buf, size) == Wasm_Module_AoT) {
 #if WASM_ENABLE_AOT != 0
         module_common = (WASMModuleCommon *)aot_load_from_aot_file(
             buf, size, error_buf, error_buf_size);
-        return register_module_with_null_name(module_common, error_buf,
-                                              error_buf_size);
+        module_common = register_module_with_null_name(module_common, error_buf,
```

Comment:
I presume the registration could be moved down to a common code?

Reply:
good point, I didn't notice it